### PR TITLE
Components: refactor `NavigatorButton` to pass `exhaustive-deps`

### PR DESCRIPTION
**Pull request by flootr**
## What?

Updates the `NavigatorButton` component to satisfy the `exhaustive-deps` eslint rule

## Why?

Part of the effort in https://github.com/WordPress/gutenberg/pull/41166 to apply `exhaustive-deps` to the Components package

## How?

Add the `attributeName` prop & `escapedName` to the `useCallback` dependency array.

## Testing Instructions

1. From your local Gutenberg directory, run `npx eslint --rule 'react-hooks/exhaustive-deps: warn' packages/components/src/navigation/index.js`
2. Confirm that the linter returns no errors
3. Confirm unit tests still pass
4. Run Storybook locally, confirm the components stories and/or docs still work as expected

### CODE REVIEW

**Review by chad1008 — APPROVED** [review]
Code changes look good to me, Storybook and unit tests all going well! @flootr can you please add a CHANGELOG entry before merging? 🙂

**Comment by flootr** [comment]
@chad1008 added via https://github.com/WordPress/gutenberg/pull/42051/commits/143701900ac381ec41e9ce0ee1f330d5aea5d4ab 👍🏻

**Review by ciampo — APPROVED** [review]
LGTM 🚀 

Let's merge this once conflicts have been solved and CI checks are ✅

**Review by tyxla — APPROVED** [review]
Tested well for me as well 👍

**Pull request merged**
